File: 8_grafos_arboles/Algoritmo de Backtrack/permutation.js
function permute(numeros) {
  let dfs = function (listaNumeros, combinaciónActual, permutaciones) {
    if (!listaNumeros.length) {
      permutaciones.push(combinaciónActual); //append the permutation to the list of permutations
    }
    for (let i = 0; i < listaNumeros.length; i++) {
      dfs(listaNumeros.slice(0, i).concat(listaNumeros.slice(i + 1)), combinaciónActual.concat(listaNumeros[i]), permutaciones); //recursive call
    }
  }
  let permutaciones = [];
  dfs(numeros, [], permutaciones);
  return permutaciones;
}

/* This is an example for calling this function: */
let exercise = permute([1, 2, 3]);
console.log(exercise);

File: 8_grafos_arboles/Algoritmo de Backtrack/permutation.py
#permutation backtracking

def permute(numeros):
   def dfs(listaNumeros, combinaciónActual, permutaciones):
       if not listaNumeros:
           permutaciones.append(combinaciónActual) #append the permutation to the list of permutations
       for i in range(len(listaNumeros)):
           dfs(listaNumeros[:i]+listaNumeros[i+1:], combinaciónActual+[listaNumeros[i]], permutaciones) #recursive call
   permutaciones = []
   dfs(numeros, [], permutaciones)
   return permutaciones

""" This is an example for calling this function:"""
exercise = permute([1, 2, 3])
print(exercise)